src/wagtail_personalisation/wagtail_hooks.py

```diff
@@ -147,12 +153,35 @@ def page_listing_more_buttons(page, page_perms, is_parent=False):
                  priority=200)
 
 
+class CorrectedPagesSummaryPanel(PagesSummaryItem):
+    def get_context(self):
+        context = super(CorrectedPagesSummaryPanel, self).get_context()
+
+        pages = [page for page in Page.objects.all().specific()
+                 if (hasattr(page, 'personalisation_metadata') is False)
+                 or (hasattr(page, 'personalisation_metadata')
+                     and page.personalisation_metadata is None)
+                 or (hasattr(page, 'personalisation_metadata')
+                     and page.personalisation_metadata.is_canonical)]
```

Comment:
I feel this is a bit too much logic to include in a list comprehension. I think it will be more readable if this is written out using a for loop